autoload/highlightedyank.vim

```diff
@@ -85,6 +85,55 @@ function! s:yank_normal(count, register) abort "{{{
   endif
 endfunction
 "}}}
+function! highlightedyank#autocmd_highlight() abort "{{{
+  let view = winsaveview()
+  let region = {}
+  let region.head = getpos("'[")
+  let region.tail = getpos("']")
+  let char = getline(region.tail[1])[region.tail[2] - 1]
+  let linecontent = v:event.regcontents[len(v:event.regcontents) - 1]
+  let tailchar = linecontent[len(linecontent) - 1]
+  if char != tailchar
+    let region.tail[2] = region.tail[2] - 1
+  endif
```

Comment:
OK, now it's clear. It may be a bug of highlight module, but anyway no problem here. I will think about it later.